### DISCUSSION TITLE
Issue - DependencyInjection.ServiceProvider.ValidateService

### DISCUSSION THREAD

**Discussion by nsfuntionality** [JonPSmith/AuthPermissions.AspNetCore#53] [Q&A]
I am getting the below error, and trying to figure out the issue, I already created 1 Tenant Change Service and when i register the second tenant change service is when i get that error.

An error occurred while starting the application.
AggregateException: Some services are not able to be constructed (Error while validating the service descriptor 'ServiceType: AuthPermissions.AdminCode.ITenantChangeService Lifetime: Transient ImplementationType: NSLogistics.CarrierCode.EfCoreCode.CarrierTenantChangeService': Unable to resolve service for type 'NSLogistics.CarrierCode.EfCoreCode.CarrierDbContext' while attempting to activate 'NSLogistics.CarrierCode.EfCoreCode.CarrierTenantChangeService'.)
Microsoft.Extensions.DependencyInjection.ServiceProvider..ctor(ICollection<ServiceDescriptor> serviceDescriptors, ServiceProviderOptions options)

InvalidOperationException: Error while validating the service descriptor 'ServiceType: AuthPermissions.AdminCode.ITenantChangeService Lifetime: Transient ImplementationType: NSLogistics.CarrierCode.EfCoreCode.CarrierTenantChangeService': Unable to resolve service for type 'NSLogistics.CarrierCode.EfCoreCode.CarrierDbContext' while attempting to activate 'NSLogistics.CarrierCode.EfCoreCode.CarrierTenantChangeService'.
Microsoft.Extensions.DependencyInjection.ServiceProvider.ValidateService(ServiceDescriptor descriptor
<img width="700" alt="startup" src="https://user-images.githubusercontent.com/46289143/178043405-222125a8-11f1-41ca-aabc-8b71d75d328d.PNG">
<img width="928" alt="Error" src="https://user-images.githubusercontent.com/46289143/178043410-8ea79c39-4a73-4044-8e10-fb4184177e85.PNG">
)

**Comment by JonPSmith** [comment]
Hi @nsfuntionality,

The error is because you can only create one `ITenantChangeService` service. The `ITenantChangeService` service is there to apply changes to your tenant data when a tenant is changes - read the [Building a tenant change service](https://github.com/JonPSmith/AuthPermissions.AspNetCore/wiki/Building-a-tenant-change-service) documentation to understand what this is doing.
